### PR TITLE
HS-604: Event Service : Add gRPC Client to query Nodes based on NodeQ…

### PR DESCRIPTION
**Pull request by thomasbigger584**
HS-604: Event Service : Add gRPC Client to query Nodes based on NodeQuery

## Jira link(s)
- https://issues.opennms.org/browse/HS-604

## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.

### CODE REVIEW

File: events/traps/src/main/java/org/opennms/horizon/events/traps/EventFactory.java [hunk 1]
```diff
@@ -131,8 +134,14 @@ private boolean shouldDiscard(Event econf) {
         return false;
     }
 
-    private Optional<Integer> resolveNodeId(String location, InetAddress trapAddress, String tenantId) {
-        // TODO: Query inventory service for node id
-        return Optional.empty();
+    private Optional<Long> resolveNodeId(String location, InetAddress trapAddress, String tenantId) {
```
Comment (j-white) [inline events/traps/src/main/java/org/opennms/horizon/events/traps/EventFactory.java:137]: would it make sense to cache successful lookups?

File: events/traps/src/main/java/org/opennms/horizon/events/traps/EventFactory.java [hunk 2]
```diff
@@ -131,8 +134,14 @@ private boolean shouldDiscard(Event econf) {
...
+        String trapIpAddress = trapAddress.getHostAddress();
+        try {
+            return Optional.of(inventoryClient
+                .getNodeIdFromQuery(tenantId, trapIpAddress, location));
+        } catch (Exception e) {
+            LOG.warn("Failed to find node id for location = {}, trap address = {}, reason = {}", location, trapIpAddress, e.getMessage());
+            return Optional.empty();
```
Comment (j-white) [inline events/traps/src/main/java/org/opennms/horizon/events/traps/EventFactory.java:144]: we'll need to revisit how errors are handled for cases like these, okay for now though

File: events/traps/src/main/java/org/opennms/horizon/events/traps/EventFactory.java (hunk 1 above)
Comment (thomasbigger584) [reply events/traps/src/main/java/org/opennms/horizon/events/traps/EventFactory.java:137]: @cgorantla

Comment (cgorantla) [reply events/traps/src/main/java/org/opennms/horizon/events/traps/EventFactory.java:137]: yeah, we should cache these lookups in next iteration. Idea is to maintain cache at both ends and Cache can be updated on Events with a Streaming gRPC call.

File: events/traps/src/main/java/org/opennms/horizon/events/traps/EventFactory.java [hunk 3]
```diff
@@ -131,8 +134,14 @@ private boolean shouldDiscard(Event econf) {
         return false;
     }
 
-    private Optional<Integer> resolveNodeId(String location, InetAddress trapAddress, String tenantId) {
-        // TODO: Query inventory service for node id
-        return Optional.empty();
+    private Optional<Long> resolveNodeId(String location, InetAddress trapAddress, String tenantId) {
+        String trapIpAddress = trapAddress.getHostAddress();
```
Comment (cgorantla) [inline events/traps/src/main/java/org/opennms/horizon/events/traps/EventFactory.java:138]: `InetAddressUtils.str(trapAddress)` might be better here

**Review by cgorantla — APPROVED** [review]

**Pull request merged**